sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```
@@ -0,0 +1,238 @@
+  private static final int PARENT_ID_LENGTH = 16;
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+
+    String otTraceId = spanContext.getTraceId().toLowerBase16();
+    String xrayTraceId = TRACE_ID_VERSION + TRACE_ID_DELIMITER +
+        otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH) + TRACE_ID_DELIMITER +
+        otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+    String parentId = spanContext.getSpanId().toLowerBase16();
+    char samplingFlag = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    String traceHeader = TRACE_ID_KEY + KV_DELIMITER + xrayTraceId + TRACE_HEADER_DELIMITER +
+        PARENT_ID_KEY + KV_DELIMITER + parentId + TRACE_HEADER_DELIMITER +
+        SAMPLED_FLAG_KEY + KV_DELIMITER + samplingFlag;
+    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    try {
```

Instead of a large catch-all exception, only catch where an exception could happen. Even if we're worried about missing something, this isn't the right layer to have a catch-all.